I have an error:  "Cannot convert relative path summary.csv to an absolute file. "

``` 
* What went wrong:
Execution failed for task ':app:generateAndroidDebugPackageTree'.
> A failure occurred while executing com.getkeepsafe.dexcount.treegen.workers.LegacyWorker
   > Cannot convert relative path summary.csv to an absolute file.

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':app:generateAndroidDebugPackageTree'.

Caused by: java.lang.UnsupportedOperationException: Cannot convert relative path summary.csv to an absolute file.
	at org.gradle.api.internal.file.IdentityFileResolver.doResolve(IdentityFileResolver.java:45)
	at org.gradle.api.internal.file.AbstractFileResolver.resolve(AbstractFileResolver.java:73)
	at org.gradle.api.internal.file.AbstractFileResolver.resolve(AbstractFileResolver.java:52)
	at org.gradle.api.internal.file.DefaultFilePropertyFactory$FixedDirectory.file(DefaultFilePropertyFactory.java:109)
	at org.gradle.api.internal.file.DefaultFilePropertyFactory$DefaultDirectoryVar$3.mapValue(DefaultFilePropertyFactory.java:360)
	at org.gradle.api.internal.file.DefaultFilePropertyFactory$DefaultDirectoryVar$3.mapValue(DefaultFilePropertyFactory.java:357)
	at org.gradle.api.internal.provider.AbstractMappingProvider.get(AbstractMappingProvider.java:66)
	at com.getkeepsafe.dexcount.treegen.workers.BaseWorker.writeSummaryFile(BaseWorker.java:95)
	at com.getkeepsafe.dexcount.treegen.workers.BaseWorker.execute(BaseWorker.java:67)
	at org.gradle.workers.internal.DefaultWorkerServer.execute(DefaultWorkerServer.java:50)
	at org.gradle.workers.internal.AbstractClassLoaderWorker$1.create(AbstractClassLoaderWorker.java:47)
	at org.gradle.workers.internal.AbstractClassLoaderWorker$1.create(AbstractClassLoaderWorker.java:37)
	at org.gradle.internal.classloader.ClassLoaderUtils.executeInClassloader(ClassLoaderUtils.java:98)
	at org.gradle.workers.internal.AbstractClassLoaderWorker.executeInClassLoader(AbstractClassLoaderWorker.java:37)
	at org.gradle.workers.internal.IsolatedClassloaderWorker.execute(IsolatedClassloaderWorker.java:49)
	at org.gradle.workers.internal.IsolatedClassloaderWorkerFactory$1.lambda$execute$0(IsolatedClassloaderWorkerFactory.java:55)
	at org.gradle.workers.internal.AbstractWorker$1.call(AbstractWorker.java:44)
	at org.gradle.workers.internal.AbstractWorker$1.call(AbstractWorker.java:41)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:416)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:406)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:102)
	at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
	at org.gradle.workers.internal.AbstractWorker.executeWrappedInBuildOperation(AbstractWorker.java:41)
	at org.gradle.workers.internal.IsolatedClassloaderWorkerFactory$1.execute(IsolatedClassloaderWorkerFactory.java:49)
	at org.gradle.workers.internal.DefaultWorkerExecutor.lambda$submitWork$2(DefaultWorkerExecutor.java:200)
	at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.runExecution(DefaultConditionalExecutionQueue.java:215)
	at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.runBatch(DefaultConditionalExecutionQueue.java:164)
	at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.run(DefaultConditionalExecutionQueue.java:131)
	... 3 more

```

--- a/src/integrationTest/groovy/com/getkeepsafe/dexcount/IntegrationSpec.groovy
+++ b/src/integrationTest/groovy/com/getkeepsafe/dexcount/IntegrationSpec.groovy
@@ -41,6 +41,28 @@ class IntegrationSpec extends Specification {
     }
 
     @Unroll
+    def "completes successfully using AGP #agpVersion and Gradle #gradleVersion"() {
+        given: "an integration test project"
+        def project = projectDir(agpVersion, gradleVersion)
+
+        when:
+        def result = GradleRunner.create()
+            .withGradleVersion(gradleVersion)
+            .withProjectDir(project)
+            .withArguments(":app:countDebugDexMethods", "--stacktrace")
+            .build()
+
+        then:
+        result.task(":app:countDebugDexMethods").outcome == TaskOutcome.SUCCESS
+
+        // These version combinations were known to fail at some point.
+        // This spec serves to guard against regression.
+        where:
+        agpVersion | gradleVersion | reportedIn
+        "4.0.0"    | "6.1.1"       | "https://github.com/KeepSafe/dexcount-gradle-plugin/issues/410"
+    }
+
+    @Unroll
     def "counting AARs using AGP #agpVersion and Gradle #gradleVersion"() {
         given: "an integration test project"
         def project = projectDir(agpVersion, gradleVersion)

--- a/src/main/java/com/getkeepsafe/dexcount/plugin/AbstractLegacyTaskApplicator.java
+++ b/src/main/java/com/getkeepsafe/dexcount/plugin/AbstractLegacyTaskApplicator.java
@@ -143,7 +143,7 @@ abstract class AbstractLegacyTaskApplicator extends AbstractTaskApplicator {
         String treePath = path.replace("outputs", "intermediates") + "/tree.compact.gz";
 
         TaskProvider<LegacyGeneratePackageTreeTask> gen = getProject().getTasks().register(treeTaskName, LegacyGeneratePackageTreeTask.class, t -> {
-            t.setDescription("Generates dex method count for ${variant.name}.");
+            t.setDescription("Generates dex method count for " + variant.getName() + ".");
             t.setGroup("Reporting");
 
             t.getConfigProperty().set(getExt());

--- a/src/main/java/com/getkeepsafe/dexcount/treegen/workers/BaseWorker.java
+++ b/src/main/java/com/getkeepsafe/dexcount/treegen/workers/BaseWorker.java
@@ -54,14 +54,14 @@ public abstract class BaseWorker<P extends BaseWorker.Params> implements WorkAct
         Property<PrintOptions> getPrintOptions();
     }
 
+    private File outputDirectory = null;
+
     @Override
     public void execute() {
         try {
             PackageTree packageTree = generatePackageTree();
 
-            File outputDir = getParameters().getOutputDirectory().get().getAsFile();
-            FileUtils.deleteDirectory(outputDir);
-            FileUtils.forceMkdir(outputDir);
+            ensureCleanOutputDirectory();
 
             writeIntermediateThriftFile(packageTree);
             writeSummaryFile(packageTree);
@@ -70,6 +70,11 @@ public abstract class BaseWorker<P extends BaseWorker.Params> implements WorkAct
         } catch (IOException e) {
             throw new DexCountException("Counting dex method references failed", e);
         }
+    }
+
+    private void ensureCleanOutputDirectory() throws IOException {
+        FileUtils.deleteDirectory(getOutputDirectory());
+        FileUtils.forceMkdir(getOutputDirectory());
     }
 
     private void writeIntermediateThriftFile(PackageTree packageTree) throws IOException {
@@ -92,7 +97,7 @@ public abstract class BaseWorker<P extends BaseWorker.Params> implements WorkAct
     }
 
     private void writeSummaryFile(PackageTree packageTree) throws IOException {
-        File summaryFile = getParameters().getOutputDirectory().file("summary.csv").get().getAsFile();
+        File summaryFile = new File(getOutputDirectory(), "summary.csv");
         FileUtils.forceMkdirParent(summaryFile);
 
         String headers = "methods,fields,classes";
@@ -109,7 +114,7 @@ public abstract class BaseWorker<P extends BaseWorker.Params> implements WorkAct
     }
 
     private void writeChartFiles(PackageTree packageTree) throws IOException {
-        File chartDirectory = getParameters().getOutputDirectory().dir("chart").get().getAsFile();
+        File chartDirectory = new File(getOutputDirectory(), "chart");
         FileUtils.forceMkdir(chartDirectory);
 
         PrintOptions options = getParameters().getPrintOptions().get()
@@ -140,11 +145,18 @@ public abstract class BaseWorker<P extends BaseWorker.Params> implements WorkAct
     private void writeFullTree(PackageTree packageTree) throws IOException {
         PrintOptions options = getParameters().getPrintOptions().get();
         String fullCountFileName = getParameters().getOutputFileName().get() + options.getOutputFormat().getExtension();
-        File fullCountFile = getParameters().getOutputDirectory().file(fullCountFileName).get().getAsFile();
+        File fullCountFile = new File(getOutputDirectory(), fullCountFileName);
 
         try (BufferedWriter bw = Files.newBufferedWriter(fullCountFile.toPath())) {
             packageTree.print(bw, options.getOutputFormat(), options);
         }
+    }
+
+    private File getOutputDirectory() {
+        if (outputDirectory == null) {
+            outputDirectory = getParameters().getOutputDirectory().get().getAsFile();
+        }
+        return outputDirectory;
     }
 
     protected abstract PackageTree generatePackageTree() throws IOException;

